src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -285,15 +287,18 @@ public Parser(Compilation compilation, in JsonSourceGenerationContext sourceGene
                     INamedTypeSymbol contextTypeSymbol = (INamedTypeSymbol)compilationSemanticModel.GetDeclaredSymbol(classDeclarationSyntax);
                     Debug.Assert(contextTypeSymbol != null);
 
+                    Location contextLocation = contextTypeSymbol.Locations[0];
```

Comment:
Could this array be empty?

Reply:
No, since the context type we are processing needed to have been declared in source (at a location).

Reply:
Given that this will be used in servicing, I'd recommend introducing a defensive check, just so that we avoid surprises.

Reply:
Done